worlds/sc2/test/test_custom_mission_orders.py

```diff
@@ -101,3 +103,88 @@ def test_mini_wol_generates(self):
       self.assertEqual(flags.get(MissionFlag.Zerg, 0), 0)
       sc2_regions = set(self.multiworld.regions.region_cache[self.player]) - {"Menu"}
       self.assertEqual(len(self.world.custom_mission_order.get_used_missions()), len(sc2_regions))
+
+   def test_locked_and_necessary_item_appears_once(self):
+      # This is a filler upgrade with a parent
+      test_item = item_names.ZERGLING_METABOLIC_BOOST
+      world_options = {
+         'mission_order': 'custom',
+         'locked_items': { test_item: 1 },
+         'custom_mission_order': {
+            'test': {
+               'type': 'column',
+               'size': 5, # Give the generator some space to place the key
+               'max_difficulty': 'easy',
+               'missions': [{
+                  'index': 4,
+                  'entry_rules': [{
+                     'items': { test_item: 1 }
+                  }]
+               }]
+            }
+         }
+      }
+
+      self.generate_world(world_options)
+      test_items_in_pool = [item for item in self.multiworld.itempool if item.name == test_item]
+      self.assertEqual(len(test_items_in_pool), 1)
+      self.assertEqual(test_items_in_pool[0].classification, ItemClassification.progression)
+
+   def test_start_inventory_and_necessary_item_appears_once(self):
+      # This is a filler upgrade with a parent
+      test_item = item_names.ZERGLING_METABOLIC_BOOST
+      world_options = {
+         'mission_order': 'custom',
+         'start_inventory': { test_item: 1 },
+         'custom_mission_order': {
+            'test': {
+               'type': 'column',
+               'size': 5, # Give the generator some space to place the key
+               'max_difficulty': 'easy',
+               'missions': [{
+                  'index': 4,
+                  'entry_rules': [{
+                     'items': { test_item: 1 }
+                  }]
+               }]
+            }
+         }
+      }
+
+      self.generate_world(world_options)
+      test_items_in_pool = [item for item in self.multiworld.itempool if item.name == test_item]
+      self.assertEqual(len(test_items_in_pool), 0)
+      test_items_in_start_inventory = [item for item in self.multiworld.precollected_items[self.player] if item.name == test_item]
+      self.assertEqual(len(test_items_in_start_inventory), 1)
+      # Start inventory gets a new item so the classification is no longer progression
+      self.assertEqual(test_items_in_start_inventory[0].classification, ItemClassification.filler)
```

Comment:
I wouldn't test myself what classification of items is the start inventory. That's a core problem

What matters is:
The items exists in desired space (start inventory or pool)
If the item is not in the start inventory, it's flagged as progression